classes/Task/Miscellaneous/CheckFilesVersion.php

```diff
@@ -57,7 +57,9 @@ public function run(): int
         }
 
         foreach (['core', 'translation', 'mail'] as $type) {
-            $changedFileList[$type] = [];
+            if (!isset($changedFileList[$type])) {
```

Comment:
We know the array always has these three keys, given the method signature of `getTamperedFilesOnShop()`. We can be sure they are always set, so we could actually remove these lines.